plugins/package-managers/composer/src/main/kotlin/Composer.kt

```diff
@@ -302,6 +294,16 @@ class Composer(
     }
 }
 
+/**
+ * Return whether this String denotes a type of platform dependency, see
+ * https://getcomposer.org/doc/articles/composer-platform-dependencies.md#different-types-of-platform-packages.
+ */
+private fun String.isPlatformDependency(): Boolean =
+    this in (COMPOSER_PLATFORM_TYPES + PHP_PLATFORM_TYPES) || startsWith("ext-") || startsWith("lib-")
+
+private val COMPOSER_PLATFORM_TYPES = listOf("composer", "composer-plugin-api", "composer-runtime-api")
+private val PHP_PLATFORM_TYPES = listOf("php", "php-64bit", "php-ipv6", "php-zts", "php-debug")
```

Comment:
Should the lists be ordered alphabetically?

Reply:
I was copying these as-is from the linked page. When comparing with that, it probably makes sense to maintain the order. However, I've made these sets now to indicate that order does not matter.